assets/javascripts/discourse/components/upcoming-events-list.gjs

```diff
@@ -17,6 +17,17 @@ export const DEFAULT_TIME_FORMAT = "LT";
 const DEFAULT_UPCOMING_DAYS = 180;
 const DEFAULT_COUNT = 8;
 
+function addToResult(date, item, result) {
+  const year = date.year();
+  const month = date.month() + 1;
+  const day = date.date();
+  const monthKey = `${year}-${month}`;
```

Comment:
I think you could simply do `const monthKey = date.format("YYYY-MM");` and remove the year and month constants